### ISSUE TITLE
`createHydratedMock` and `cloneDeep` from LoDash are not loving each other

### ISSUE THREAD

**Issue by C0ZEN** [Typescript-TDD/ts-auto-mock#751]
### Subject of the issue
After migrating from `createMock` to `createHydratedMock`, I had a failing test.
I added some debuggers and I found out that nothing happened after using a [cloneDeep](https://lodash.com/docs/4.17.15#cloneDeep).
Removing the `cloneDeep` allow the test to pass as expected.
Using a spread operator is working as expected as well.
It means that there is something specific made inside the `cloneDeep` that `createHydratedMock` can't handle.

### Your environment
* ts-auto-mock version: 3.2.2
* typescript version: 4.2.4
* node version: 14.16.0
* npm version: 7.12.1
* lodash: 4.17.21

### Steps to reproduce

This is the object that should be copied (as JSON):
```
{
  "response": "dummy-response",
  "options": {
    "tts": false,
    "nonce": "",
    "disableMentions": "none",
    "files": [],
    "code": "",
    "split": false,
    "content": "",
    "embed": {},
    "allowedMentions": {
      "parse": [],
      "roles": [],
      "users": []
    },
    "reply": {}
  }
}
```
![image](https://user-images.githubusercontent.com/10194542/119241064-340f5880-bb54-11eb-91de-746cfd79ecf2.png)

### Expected behavior
The test pass and the `cloneDeep` works as expected when using `createHydratedMock`.

### Actual behavior
The test fail due to the `cloneDeep`.

**Comment by uittorio** [comment]
Hi @C0ZEN we are getting there!!!

Do you have an isolated example of the issue?

Which object are you trying to clone and at which point?

I will try to use createHydratedMock in combination with cloneDeep but it will help having the specific example! 

If you dont have time I'll try to find it from your repository!

**Comment by C0ZEN** [comment]
@uittorio hello, I am reaching the end of my road to hydrated mocks ;)
You can see here https://github.com/Sonia-corporation/sonia-discord/pull/1383/checks?check_run_id=2647456167.
This is the PR with the migration and the failing tests are related to the cloneDeep usage.

**Comment by uittorio** [comment]
So, I've done some research. The interface has recursion. CloneDeep will iterate through all the values of the interface it will reach a Maximum call stack size exceeded. If you wrap the clone deep code in a try and catch you'll see the error.  Why is this happening with createHydratedMock and not createMock? Because the recursion is part of the not required interface.

**Comment by uittorio** [comment]
I've created an example without ts auto mock

**Comment by uittorio** [comment]
If you instead use a class with a getter it's all good  because the javascript converted has enumerable: false

**Comment by Pmyl** [comment]
Yeah, cloneDeep for recursive interface would break it. It may be not break for `createMock` because it was defaulting some cases of `Type | null` to null so there is no recursion.

I guess it's dangerous to use cloneDeep on recursive interfaces in general, contracts like "this recursion is never infinite" cannot be defined in an interface, creating cases like ours where our implementation of that interface is correct but the code fails.
I checked and cloneDeep doesn't have any way to limit the depth, that would have been useful.
Maybe we have to create something ourselves to limit the max recursion depth? @uittorio

**Comment by uittorio** [comment]
I think is a good idea to limit recursion, however recursive getter in classes works with cloneDeep! I was wondering if we are doing anything wrong!

**Comment by Pmyl** [comment]
you're right, we should find a way to ensure it works the same way, sorry @C0ZEN this may take a bit :(

**Comment by C0ZEN** [comment]
As usual guys, I am not rushing and this is an open-source project, so take the time you need and anyhow I have a fallback option with the spread operator (even though cloneDeep is better than spread in my XP).
Thanks!
